client/hostinfo_unix.cpp

```diff
@@ -1584,6 +1584,47 @@ int HOST_INFO::get_memory_info() {
     return 0;
 }
 
+#if LINUX_LIKE_SYSTEM
+// gather libc version from the system installed ldd binary
+// a usual first line looks like: ldd (Debian GLIBC 2.24-11+deb9u1) 2.24
+// where the last part is copied to version and the string between parenthesis
+// is copied to extra_info
+//
+// return BOINC_SUCCESS if at least version could be found (extra_info may remain empty)
+// return ERR_NOT_FOUND if ldd couldn't be opened or no version information was found
+//
+int get_libc_version(string& version, string& extra_info) {
+    char buf[1024];
+    string strbuf;
+    FILE* f = popen("PATH=/usr/bin:/bin:/usr/local/bin ldd --version 2>&1", "r");
+    if (f) {
+        fgets(buf, sizeof(buf), f);
+        if (0 > pclose(f)) {
```

Comment:
No, still doesn't work right. Without `ldd`:
```
30-Jan-2018 23:33:09 [---] [libc detection] gathered: found,
30-Jan-2018 23:33:09 [---] OS: Linux LinuxMint: Linux Mint 17.3 Rosa [3.13.0-123-generic|libc found]
```
`pclose` returns `-1` only in case of internal error, otherwise it's the exit status as in `waitpid` and needs to be examined with `W*` macros.

Even with `ldd` in place the exit status is `WIFEXITED=1` and `WEXITSTATUS=141`. I don't know why that is. Maybe because you close the pipe before reading everything.

Reply:
Yep. `WEXITSTATUS=141` came from closing the pipe too early. This seems to work:
```
 int get_libc_version(string& version, string& extra_info) {
-    char buf[1024];
+    char buf[1024] = "";
     string strbuf;
     FILE* f = popen("PATH=/usr/bin:/bin:/usr/local/bin ldd --version 2>&1", "r");
     if (f) {
-        fgets(buf, sizeof(buf), f);
-        if (0 > pclose(f)) {
+        char* retval = fgets(buf, sizeof(buf), f);
+        strbuf = buf;
+        while (fgets(buf, sizeof(buf), f)) {
+            // consume output to allow command to exit gracefully
+        }
+        int status = pclose(f);
+        if (!retval || status == -1 || !WIFEXITED(status) || WEXITSTATUS(status)) {
             return ERR_NOT_FOUND;
         }
-        strbuf = (string)buf;
         strip_whitespace(strbuf);
```

Or you could have the code check that the first line matches the expected format.

Reply:
I suspected that closing the pipe early is going to cause problems but I hadn't had time to look into it. Thanks for testing. I'm going to adapt your code into this PR.

Reply:
I'm having difficulty following github's intertwined changes and reviews, but it looks to me as if the latest change is 5b94cf8defeca4e157ae8c92c4e05c797f042fc2 two days ago (which doesn't show in this timeline). @JuhaSointusalo, does this satisfy your most recent change request?